Tasks/DeployAzureResourceGroup/SPNCreation.ps1

```diff
@@ -13,9 +16,9 @@ param
 #Initialize
 $ErrorActionPreference = "Stop"
 $VerbosePreference = "SilentlyContinue"
-$userName = $env:USERNAME
+#$userName = $env:USERNAME
 $newguid = [guid]::NewGuid()
-$displayName = [String]::Format("VSO.{0}.{1}", $userName, $newguid)
+$displayName = [String]::Format("VSO.{0}.{1}", $appName, $newguid)
```

Comment:
If we agree on making this optional, then the displayname should use username as it does today if appname is null/empty.